test/cli/testutils.py

```diff
@@ -70,9 +78,31 @@ def __lookup_cppcheck_exe():
     return exe_path
 
 
+def __copy_and_prepare_cppcheck(tmpdir):
```

Comment:
I would suggest that you make this function public so tests calls it directly and only once. Your test will execute function cppcheck() twice and therefore it executes this __copy_and_prepare_cppcheck twice also..